dockerfile

```diff
@@ -0,0 +1,112 @@
+FROM ubuntu:20.04
+
+SHELL [ "/bin/bash", "-c" ]
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Make sure we're starting with an up-to-date image
+RUN apt-get update && \
+    apt-get upgrade -y && \
+    apt-get autoremove -y --purge && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+# To mark all installed packages as manually installed:
+#apt-mark showauto | xargs -r apt-mark manual
+
+RUN apt-get update && \
+    apt-get install -y \
+        apt-transport-https \
+        ccache \
+        g++-10 \
+        gcc \
+        gcc-10 \
+        git \
+        gnupg \
+        help2man \
+        libbz2-dev \
+        libcurl4-gnutls-dev \
+        libfuse-dev \
+        libjson-perl \
+        libkrb5-dev \
+        libpam0g-dev \
+        libssl-dev \
+        libxml2-dev \
+        lsb-release \
+        lsof \
+        make \
+        ninja-build \
+        odbc-postgresql \
+        postgresql \
+        python3 \
+        python3-dev \
+        python3-pip \
+        python3-distro \
+        python3-jsonschema \
+        python3-packaging \
+        python3-psutil \
+        python3-pyodbc \
+        python3-requests \
+        sudo \
+        super \
+        unixodbc-dev \
+        wget \
+        zlib1g-dev \
+    && \
+    pip3 --no-cache-dir install lief && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+RUN wget -qO - https://packages.irods.org/irods-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://packages.irods.org/apt/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/renci-irods.list && \
+    wget -qO - https://core-dev.irods.org/irods-core-dev-signing-key.asc | apt-key add - && \
+    echo "deb [arch=amd64] https://core-dev.irods.org/apt/ $(lsb_release -sc) main" | tee /etc/apt/sources.list.d/renci-irods-core-dev.list
+
+RUN apt-get update && \
+    apt-get install -y \
+        'irods-externals*' \
+    && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+RUN update-alternatives --install /usr/local/bin/gcc gcc /usr/bin/gcc-10 1 && \
+    update-alternatives --install /usr/local/bin/g++ g++ /usr/bin/g++-10 1 && \
+    hash -r
+
+ARG cmake_path="/opt/irods-externals/cmake3.21.4-0/bin"
+ENV PATH=${cmake_path}:$PATH
+
+
+
+# Build boost 1.81
+RUN wget https://boostorg.jfrog.io/artifactory/main/release/1.81.0/source/boost_1_81_0.tar.bz2 && \
+    tar xaf boost_1_81_0.tar.bz2
```

Comment:
So this project requires newer versions of Boost.

Are there other libraries we need which aren't provided via the iRODS externals packages? If so, let's capture those libraries (and Boost) in an issue. The issue can talk about enabling compilation of this project using only the externals packages (future work).

Reply:
Happily no, we don't need any other libraries other than newer boost.